src/utils.js

```diff
@@ -0,0 +1,23 @@
+export function kindof(obj) {
+    if (obj === undefined) return 'undefined';
```

Comment:
@stavalfi Moreover if you use lodash, `isNil` would shorten it :)

Reply:
I think it's good to add third parties libs that will help us but libs like lodash are problematic because the user which will use our lib will probably have lodash also and it will cause 
1. dependency conflicts and will have no solution unless using our lodash version.
2. he will use his version and our version using webpack and as a result, he will have duplicate functions in his bundles.

As for those reasons, I disagree to add lodash or underscore as dependencies. 
---

If you think that some of our existing dependencies will cause the user some dependencies problems, feel free to open issue for each of them. (we need to remove kindof because we copied his implementation to utils)

Reply:
> @stavalfi I'll suggest
> `if (obj === undefined || obj === null) return obj;`

it's unlikely but if `obj` value equals one of the cases, we will have a very hard to find a bug.
I think that you are mixing the type and value of a variable together. 

Do you agree?